### ISSUE TITLE
Epic: Build Atlas Agent Evaluation Lab for reproducible LangGraph experiments, regression gates, and DigitalOcean Managed Agents

### ISSUE THREAD

**Issue by CarawayLabs** [Caraway-Labs/one-health-lyme-gap-atlas-shared-python#10]
## Issue Type
Epic

## Product intent
Build an **Agent Evaluation Lab** that becomes the common experimentation, replay, comparison, and regression-gating system for Atlas AI agents. The Lab should answer a question that becomes critical as Atlas adds Research Assistant, Ask Atlas, Evidence Summarization, County Intelligence, Dataset Discovery, Knowledge-Graph Agent, Data Pipeline Guardian, and PR Architecture Reviewer:

> Is the new graph/model/prompt/tool configuration actually better, safer, faster, and cheaper than the current one — on the same versioned evidence?

The Lab is not another end-user agent. It is an engineering/evaluation platform that runs controlled agent experiments and produces evidence for promotion decisions.

## Current-state grounding (reviewed from `main` and current backlog)
Atlas already has important pieces that the Lab should unify rather than replace:
- `one-health-lyme-gap-atlas-knowledge-graph/evals/chat-v1.jsonl` contains a versioned evaluation corpus with surveillance, vector/pathogen, environment, diagnostics, conflicting-evidence, no-evidence, medical-safety, prompt-injection, and paraphrase cases.
- Multiple current agent epics explicitly require Arize/Phoenix traces, versioned eval datasets, experiments, trajectory evals, regression cases, latency, token/cost, provenance, abstention, and production failure review.
- The data ML epic already calls for a reproducible experiment runner/model registry pattern, leakage-safe datasets, comparison artifacts, monitoring, and reproducibility.
- `one-health-lyme-gap-atlas-shared-python` already owns typed shared infrastructure plus redacted JSON logging and optional OTLP tracing.
- API and data services already consume the shared package and configure OTLP tracing.
- DigitalOcean Managed Agents exposes hosted LangGraph execution, exact commit pinning, triggers, session telemetry, checkpoints/forks for supported coding agents, and Action Gateway tools. Custom LangGraph graphs remain ordinary repositories/applications that can also run locally.

Today, evaluation requirements are distributed across agent epics. The Lab should provide common contracts and runners so each agent defines **domain-specific datasets/evaluators**, while shared experiment plumbing is implemented once.

## Repository ownership decision
Use `one-health-lyme-gap-atlas-shared-python` as the coordination home for cross-agent evaluation contracts and the Lab app, while preserving the current shared-library rule that application orchestration does not live inside `src/lyme_gap_atlas_shared`.

Recommended shape:

```text
src/lyme_gap_atlas_shared/
  evaluation/              # reusable typed contracts/utilities only

agents/
  evaluation-lab/          # deployable app/orchestration
    langgraph.json
    pyproject.toml
    src/...
    specs/...
```

The deployable Lab graph should use `FRAMEWORK_SUBDIR` on DigitalOcean Managed Agents. Domain agent repositories keep their own authoritative fixtures where appropriate and expose them through a versioned adapter/manifest rather than copying everything into this repo.

## Business goals
1. Make agent quality changes measurable instead of subjective.
2. Provide one reproducible experiment vocabulary across Atlas agents.
3. Catch regressions in groundedness, tool behavior, safety, provenance, latency, and cost before promotion.
4. Make production failures easy to convert into versioned regression cases.
5. Compare models, prompts, graph topology, tools, thresholds, and retrieval strategies on identical fixtures.
6. Preserve enough run metadata to reproduce historical results.
7. Separate deterministic evaluators from model-based judges and measure judge reliability.
8. Support both local/CI experiments and hosted DigitalOcean Managed Agents experiments without rewriting agent code.
9. Create a strong public technical artifact demonstrating disciplined agent engineering/evaluation, not just agent demos.

## Non-goals
- No autonomous model/prompt promotion to production in v1.
- No training/fine-tuning platform.
- No replacement for domain-specific tests, deterministic unit/integration tests, or ML model training infrastructure.
- No single universal score that hides safety or critical dimension regressions.
- No hidden chain-of-thought capture as an evaluation artifact.
- No unrestricted production-data replay; datasets must follow each agent's privacy/governance rules.

## Core architecture
```text
Versioned Evaluation Dataset / Manifest
              ↓
     Experiment Configuration
       ├─ agent adapter
       ├─ graph/code SHA
       ├─ model/provider
       ├─ prompt/config
       ├─ tool versions
       └─ budgets
              ↓
      LangGraph Lab Orchestrator
       ├─ materialize cases
       ├─ execute candidate A/B/... 
       ├─ capture normalized run traces/results
       ├─ run deterministic evaluators
       ├─ run approved LLM judges
       ├─ calculate latency/token/cost/safety metrics
       └─ persist experiment artifacts
              ↓
       Comparison / Gate Engine
       ├─ per-case diffs
       ├─ aggregate metrics
       ├─ slice metrics
       ├─ regression detection
       └─ promotion recommendation
              ↓
     Human review / CI / report
```

## LangGraph composition
The Lab itself should be an explicit graph, not a free-form evaluator agent. Required stages:
1. `load_experiment_spec`
2. `validate_dataset_manifest`
3. `resolve_candidate_configs`
4. `plan_case_execution`
5. fan-out `execute_case_candidate`
6. `normalize_run_artifact`
7. `run_deterministic_evaluators`
8. conditional `run_llm_judges` only for configured semantic dimensions
9. `aggregate_case_results`
10. `compute_slices_and_costs`
11. `compare_candidates`
12. `apply_regression_gates`
13. `generate_experiment_report`
14. `persist_traceable_artifacts`
15. `finalize`

Parallelism must be bounded by explicit concurrency/rate/cost limits.

## Required Lab state
Create typed/versioned `EvaluationLabState` with at minimum:

```text
- schema_version
- experiment_id
- experiment_spec_version
- dataset_manifest_ref/version/hash
- candidate_configs[]
- evaluator_configs[]
- judge_configs[]
- case_plan[]
- execution_results[]
- normalized_run_artifacts[]
- deterministic_scores[]
- judge_scores[]
- per_case_results[]
- slice_results[]
- aggregate_results[]
- candidate_comparisons[]
- regression_gate_results[]
- latency/token/cost summaries
- trace/run/session refs
- budget/concurrency state
- errors[]
- final_status
```

Do not store model hidden reasoning. Store observable messages/tool calls/evidence/output/configuration only where permitted by the agent's data policy.

## Common evaluation contracts
The shared library should define stable typed contracts for:
- dataset manifest;
- evaluation case;
- expected outcome/ground truth;
- candidate configuration;
- normalized agent run artifact;
- tool-call/trajectory summary;
- evaluator definition/result;
- judge definition/result;
- experiment run/result;
- comparison result;
- regression gate result;
- human review/adjudication;
- failure-to-regression provenance.

Domain agents may extend these contracts with domain payloads through explicit typed extension points, not arbitrary dictionaries where avoidable.

## Evaluation philosophy
- Prefer deterministic evaluators whenever the criterion is mechanically checkable.
- Use LLM judges for semantic dimensions that cannot be reliably scored deterministically.
- Version judge prompts/models and test them against human-labeled calibration sets.
- Never let one overall composite score override a hard safety/governance failure.
- Report slices by dataset category, agent, tool path, repository, geography, safety class, or other domain-relevant segments.
- Separate answer/output quality from trajectory/tool quality, reliability, latency, and cost.

## Initial shared metric dimensions
Common dimensions should support:
- task success/correct final outcome;
- groundedness/evidence support;
- citation/provenance fidelity;
- unsupported claim rate;
- abstention/refusal correctness;
- tool-selection correctness;
- tool argument validity;
- trajectory/stop-condition quality;
- policy/governance compliance;
- deterministic contract/schema correctness;
- latency;
- tool/model call count;
- input/output tokens;
- estimated/actual cost;
- reliability/error/retry rate.

Each agent chooses the subset that applies plus domain-specific metrics.

## Storage / reproducibility
Persist or export enough metadata to reproduce a run:
- experiment/dataset/case IDs;
- code SHA;
- spec/graph/prompt/tool/config versions;
- model/provider/model identifier;
- evaluator/judge versions;
- input/evidence snapshot references or immutable hashes where policy permits;
- run/trace/session IDs;
- outputs/scores;
- timestamps and runtime package/environment metadata.

Do not persist secrets. Do not duplicate governed raw data when stable immutable references are sufficient.

Snowflake may be used for queryable experiment metadata/results because Atlas already uses it as the governed analytical store, but the spec must define a clean repository/service boundary and avoid forcing every local unit test to require Snowflake.

## Arize/Phoenix / OTEL integration
The Lab should consume/export OpenTelemetry-compatible trace identifiers and support Arize/Phoenix experiment/trace workflows already planned by current agent epics. Arize/Phoenix is the observability/evaluation surface, not the sole source of truth for versioned fixtures/specs. Repository-owned manifests and immutable run metadata must remain sufficient to understand what was evaluated.

## DigitalOcean Managed Agents use
The Lab should be deployable as a custom LangGraph framework agent:
- exact `FRAMEWORK_REPO_SHA`;
- `FRAMEWORK_SUBDIR` for the Lab app;
- managed secrets;
- narrow Action Gateway tools;
- manual/cron execution for benchmark suites as approved;
- bounded concurrency/token/cost/time budgets;
- session telemetry correlated into experiment records.

Use hosted runs to compare the behavior/cost of agents in the same environment in which some production experiments operate, while retaining local/CI execution for deterministic development loops.

## Spec-driven implementation contract
Story 1 must create `agents/evaluation-lab/specs/v1/` (or equivalent) containing:
- `product.md`
- `architecture.md`
- `graph.md`
- `state.schema.json`
- `dataset.schema.json`
- `run-artifact.schema.json`
- `evaluator.schema.json`
- `experiment.schema.json`
- `comparison-and-gates.md`
- `storage.md`
- `observability.md`
- `security-and-data-policy.md`
- `deployment.md`
- `integration-contracts.md`
- `open-questions.md`

Later stories implement against these contracts and update them explicitly when behavior changes.

## Child-story sequence
1. Define Evaluation Lab spec, common evaluation ontology/contracts, LangGraph state, storage, security, and promotion policy.
2. Implement reusable typed evaluation contracts plus dataset/fixture manifests and validation tooling.
3. Implement agent adapters and normalized run-artifact capture for local/LangGraph/hosted workloads.
4. Implement deterministic evaluator framework and calibrated/versioned LLM judge framework.
5. Implement LangGraph experiment orchestration, bounded parallel execution, candidate comparison, slicing, and statistical/variance handling.
6. Persist/query experiment metadata/results and integrate OTEL + Arize/Phoenix trace/experiment linking.
7. Add CI/regression gates, human adjudication, and production-failure-to-regression workflows; onboard Guardian and PR Reviewer as first workloads.
8. Deploy Evaluation Lab to DigitalOcean Managed Agents with scheduled/manual experiment modes, permissions, budgets, reporting, and runbook.

## Definition of done
- At least two different Atlas agent workloads use the shared Lab contracts/runners.
- The same dataset can compare two candidate configurations reproducibly.
- Every result is tied to code/model/prompt/tool/evaluator/dataset versions.
- Deterministic safety/governance failures can block a candidate regardless of aggregate quality score.
- LLM judge behavior is versioned/calibrated against human labels.
- Per-case, aggregate, and slice results plus latency/cost are available.
- Confirmed field failures can be converted into versioned regression cases.
- Local/CI and DigitalOcean-hosted experiments use the same normalized result contract.
- Arize/Phoenix/OTEL traces are linkable to experiment results without making external observability the only record of the experiment.
- No automatic production promotion occurs without explicit human workflow.

**Comment by CarawayLabs** [comment]
## Created child stories / implementation sequence

1. #11 — Define Evaluation Lab spec, evaluation ontology/contracts, LangGraph state, storage, security, and promotion policy
2. #12 — Implement reusable evaluation contracts, versioned dataset manifests, fixtures, validation, and hashing utilities
3. #13 — Implement agent adapters and normalized run-artifact capture for local, LangGraph, and hosted workloads
4. #14 — Implement deterministic evaluator framework and calibrated, versioned LLM judge framework
5. #15 — Implement Evaluation Lab LangGraph experiment orchestration, bounded parallelism, candidate comparison, slicing, and variance analysis
6. #16 — Persist/query experiment results with Snowflake-ready storage plus OTEL and Arize/Phoenix trace linking
7. #17 — Add CI regression gates, human adjudication, failure-to-regression workflow, and onboard Guardian + PR Reviewer
8. #18 — Deploy Evaluation Lab to DigitalOcean Managed Agents with manual/scheduled experiments, permissions, budgets, reporting, and runbook

Each child story includes `Parent`, `Implementation Order`, explicit prerequisites, acceptance criteria, and Codex completion evidence. This sequence is authoritative for v1 unless the versioned Lab spec is intentionally revised.
